frontend/src/context/CartContext.tsx

```diff
@@ -0,0 +1,184 @@
+import { createContext, useContext, useState, useEffect, ReactNode } from 'react';
+import axios from 'axios';
+import { api } from '../api/config';
+
+interface Product {
+  productId: number;
+  name: string;
+  description: string;
+  price: number;
+  imgName: string;
+  sku: string;
+  unit: string;
+  supplierId: number;
+  discount?: number;
+}
+
+interface CartItem {
+  cartItemId: number;
+  cartId: number;
+  productId: number;
+  quantity: number;
+  addedAt: string;
+  product: Product;
+}
+
+interface Cart {
+  cartId: number;
+  sessionId: string;
+  createdAt: string;
+  updatedAt: string;
+  items: CartItem[];
+}
+
+interface CartContextType {
+  cart: Cart | null;
+  loading: boolean;
+  addToCart: (productId: number, quantity: number) => Promise<void>;
+  updateQuantity: (cartItemId: number, quantity: number) => Promise<void>;
+  removeItem: (cartItemId: number) => Promise<void>;
+  clearCart: () => Promise<void>;
+  refreshCart: () => Promise<void>;
+  getTotalItems: () => number;
+  getTotalPrice: () => number;
+}
+
+const CartContext = createContext<CartContextType | undefined>(undefined);
+
+const CART_SESSION_KEY = 'octocat_cart_session';
+
+const getSessionId = (): string => {
+  let sessionId = localStorage.getItem(CART_SESSION_KEY);
+  if (!sessionId) {
+    sessionId = `session_${Date.now()}_${Math.random().toString(36).substring(2, 15)}`;
```

Comment:
The session ID generation uses `Math.random()` which is not cryptographically secure. While this is acceptable for a demo application, for production use, consider using `crypto.randomUUID()` or the Web Crypto API to generate unpredictable session identifiers. This prevents potential session hijacking if an attacker can predict session IDs.

Example:
```typescript
sessionId = `session_${Date.now()}_${crypto.randomUUID()}`;
```
```suggestion
    sessionId = `session_${Date.now()}_${crypto.randomUUID()}`;
```